sdks/python/apache_beam/testing/benchmarks/nexmark/nexmark_launcher.py

```diff
@@ -0,0 +1,199 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""Nexmark launcher.
+
+The Nexmark suite is a series of queries (streaming pipelines) performed
+on a simulation of auction events. The launcher orchestrates the generation
+and parsing of streaming events and the running of queries.
+
+Model
+- Person: Author of an auction or a bid.
+- Auction: Item under auction.
+- Bid: A bid for an item under auction.
+
+Events
+ - Create Person
+ - Create Auction
+ - Create Bid
+
+Queries
+- Query0: Pass through (send and receive auction events).
+
+Usage
+- DirectRunner
+python nexmark_launcher.py \
+    --query/q <query number> \
+    --project <project id> \
+    --streaming \
+
+- DataflowRunner
+python nexmark_launcher.py \
+    --query/q <query number> \
+    --project <project id> \
+    --streaming \
+    --sdk_location <apache_beam tar.gz> \
+    --staging_location=gs://... \
+    --temp_location=gs://
+
+"""
+from __future__ import print_function
+
+import argparse
+import logging
+import sys
+
+import apache_beam as beam
+from apache_beam.options.pipeline_options import GoogleCloudOptions
+from apache_beam.options.pipeline_options import PipelineOptions
+from apache_beam.options.pipeline_options import SetupOptions
+from apache_beam.options.pipeline_options import StandardOptions
+from apache_beam.options.pipeline_options import TestOptions
+
+from nexmark_util import Command
+from queries import query0
+
+logging.basicConfig(level=logging.INFO,
+                    format='(%(threadName)-10s) %(message)s')
+
+# Reserved configuration to run these benchmark suite.
+PUBSUB_PROJECT = 'google.com:clouddfe'
+PUBSUB_SUBSCRIPTION = 'nexmark-benchmark'
+PUBSUB_TOPIC = 'nexmark-benchmark'
+
+# TODO(mariagh): Make comprehensive and large
+EVENT_COUNT_PER_TYPE = 3
+PERSON_EVENT = 'p12345,maria,maria@maria.com,1234-5678-9012-3456,sunnyvale,CA,1528098831536' # pylint: disable=line-too-long
+AUCTION_EVENT = 'a12345,car,2012 hyundai elantra,20K,15K,1528098831536,20180630,maria,vehicle' # pylint: disable=line-too-long
+BID_EVENT = 'b12345,maria,354,1528098831536'
+
+
+def parse_args():
+  parser = argparse.ArgumentParser()
+
+  parser.add_argument('--query', '-q',
+                      type=int,
+                      action='append',
+                      required=True,
+                      choices=[0, 1],
+                      help='Query to run')
+
+  parser.add_argument('--subscription',
+                      type=str,
+                      help='Pub/Sub subscription to read from')
+
+  parser.add_argument('--topic',
+                      type=str,
+                      help='Pub/Sub topic to read from')
+
+  args, pipeline_args = parser.parse_known_args()
+  options = PipelineOptions(pipeline_args)
+  print('args, pipeline_args:', args, pipeline_args)
```

Comment:
Use logging instead of print statement please. The reason is, logging is configurable, includes additional information (such as logging levels).

Reply:
Done.